front-end/src/views/function/components/AddFunc.vue

```diff
@@ -0,0 +1,242 @@
+    }
+  },
+  data() {
+    return {
+      file:'',
+      form: {
+        functionName: '',
+        inputs: [''],
+        output: '',
+        logTopic: '',
+        className: '',
+        runtime: 'JAVA'
+      },
+      rules:{
+        functionName:[
+          { required: true, message: '请输入FunctionName', trigger: 'blur' }
+        ],
+        output:[
+          { required: true, message: '请输入Output', trigger: 'blur' }
+        ],
+        logTopic:[
+          { required: true, message: '请输入Log-topic', trigger: 'blur' }
+        ],
+        className:[
+          { required: true, message: '请输入ClassName', trigger: 'blur' }
+        ]
+      }
+    }
+  },
+  methods: {
+    onSubmit() {
+      this.$refs.addForm.validate(async valid =>{
+        if(!valid) return
+        if(!this.file){
+          this.$message('还没添加文件')
+          return
+        }
+        const functionConfig = JSON.parse(JSON.stringify(this.form).replace(/logTopic/g,'log-topic'))
+        const functionName = functionConfig.FunctionName
+        delete functionConfig.functionName
+        console.log(functionConfig)
+        const data = new FormData()
+        data.append('data', this.file)
+        data.append('functionConfig',new Blob([JSON.stringify(functionConfig)], { type: 'application/json' }))
+        console.log(data)
+        this.$confirm('即将添加函数','提示',{
+          confirmButtonText: '确定',
+          cancelButtonText: '取消',
+          type: 'warning'
+        }).then(() => {
+          create(functionName, data)
+            .then((res) => {
+              this.$notify({
+                type: 'success',
+                message: '添加成功'
+              })
+              this.$refs.addForm.resetFields()        //清空已输入的数据
+              this.file = ''                          //清空文件
+              this.$refs.file.clearFiles()            //清空已上传文件列表
+              this.$refs.addFuncDrawer.handleClose()  //关闭drawer
+              this.refresh()                          //刷新table上的数据
+            })
+        }).catch(() => {
+          this.$notify({
+            type: 'info',
+            message: '添加失败'
+          })
+        })
+      })
+      // console.log(JSON.parse(JSON.stringify(this.form)))
+    },
+    addInput(){
+      this.form.inputs.push('')
+    },
+    deleteInput(index){
+      this.form.inputs.splice(index,1)
+    },
+    resetField(formName){
+      this.$refs[formName].resetFields()
+    },
+    getFile(file){
+      this.file = file.raw
+    }
+  }
+}
+</script>
+<style scoped>
+.upload ::v-deep(.el-upload){
+  width: 100%;
+}
+.upload ::v-deep(.el-upload-dragger){
+  width: 100%;
+}
+.el-form-item--mini.el-form-item, .el-form-item--small.el-form-item{
+  margin-bottom: 0px;
+}
+.addFunc ::v-deep(.el-form-item__error){
+  position: relative;
+  float: left;
+}
+.el-drawer ::v-deep(.el-drawer__body){
+  overflow: auto!important;
+}
+/*.el-drawer rtl ::v-deep(.el-drawer__container ::-webkit-scrollbar){*/
+/*  display: none;*/
+/*}*/
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834759)
```suggestion
  /*}*/
```